js/samples/ember-sample/app/components/reader-view.js

```diff
@@ -59,4 +59,38 @@ export default class ReaderViewComponent extends Component {
             }
         }
     }
+
+    @action
+    async handleLaunchImmersiveReader(sampleId) {
+        await this.getTokenAndSubdomainAsync();
+        var langElement = $('#lang_' + sampleId);
+        var lang = sampleId == 'DisableLanguageDetection' ? langElement.val() : 'en';
+
+        const data = {
+            title: $('#title' + sampleId).text().trim(),
+            chunks: [{
+                content: $('#text' + sampleId).text().trim(),
+                lang: lang
+            }]
+        };
+
+        let options = {};
+
+        options.disableLanguageDetection = sampleId === 'DisableLanguageDetection';
+        options.disableGrammar = sampleId === 'DisableGrammar';
+        options.disableTranslation = sampleId === 'DisableTranslation';
+        options.parent = sampleId === 'Parent' && document.getElementById("checkboxParent").checked ? document.getElementById('parentDiv') : null;
+
+        if (this.token === null) {
```

Comment:
don't think this IF statement for token here or above is needed anymore since token is now fetched automatically

Reply:
makes sense, IF removed